lib/collection/src/shards/transfer/stream_records.rs

```diff
@@ -37,7 +41,13 @@ pub(super) async fn transfer_stream_records(
             )));
         };
 
-        replica_set.proxify_local(remote_shard).await?;
+        let points_count = replica_set
+            .proxify_local(remote_shard)
+            .await?
+            .wrapped_shard
+            .read_filtered(None)?
+            .len();
```

Comment:
I'd prefer to do these two things separate. Maybe we can utilize `replica_set.count_local` for it.

```suggestion
        replica_set.proxify_local(remote_shard).await?;
            
        let points_count = replica_set.count_local(...).await?;
```

Reply:
Fixed.